pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/parser/GroovyShellDecoratorImpl.java

```diff
@@ -33,15 +46,34 @@ public void configureCompiler(@CheckForNull final CpsFlowExecution execution, Co
         cc.addCompilationCustomizers(new CompilationCustomizer(CompilePhase.SEMANTIC_ANALYSIS) {
             @Override
             public void call(SourceUnit source, GeneratorContext context, ClassNode classNode) throws CompilationFailedException {
-                // TODO: workflow-cps-plugin CpsFlowExecution.parseScript() should be passing in CodeSource
-                // to help us determine that that is a user-written script
-                // Commenting out for findbugs
-                // CodeSource cs = classNode.getCompileUnit().getCodeSource();
-                // if (<< cs comes from Jenkinsfile>>) {
-
-                // but until that happens,
-                // Using getNameWithoutPackage to make sure we don't end up executing without parsing when an inadvertent package name is put in the Jenkinsfile.
-                if (classNode.getNameWithoutPackage().equals(Converter.PIPELINE_SCRIPT_NAME)) {
+                boolean doModelParsing = false;
+
+                // First, we'll parse if this is WorkflowScript, i.e., a Jenkinsfile, or it's an autogenerated script
+                // with no package, i.e., Script1
+                if (classNode.getNameWithoutPackage().equals(Converter.PIPELINE_SCRIPT_NAME) ||
+                        (classNode.getPackageName() == null && classNode.getNameWithoutPackage().matches("Script\\d+"))) {
```

Comment:
Going to blow up when someone fixes [JENKINS-31838](https://issues.jenkins-ci.org/browse/JENKINS-31838). Find a better way. `CodeSource`?

Reply:
This would bust the line above too, right? Which we've had in place for ages...hrm. Lemme think on it for a bit.